js-exercises/object-utils/objectUtils.js

```diff
@@ -0,0 +1,59 @@
+// ! code may break didn't consider many edge cases
+
+// ? 1 If Object has Array/function in value
+// ? 2 If user does a arithmetic operation in callBackFunction on string value
+// ? 3 if user want to access/modify - not iteratable for not modifiable key value
+
+// ? Thought of using TRY CATCH block to throw error in case of Error.
+// # Is it a good approach @ ARFAT
+
+
+const errorCheck = (obj) => {
+  if (obj === null) throw new Error('received NULL -- expecting OBJECT ')
+  if (typeof obj !== 'object') throw new Error(`received ${typeof obj} -- expecting OBJECT `);
```

Comment:
No need to check individually.

Reply:
![Screenshot from 2021-01-07 18-06-11](https://user-images.githubusercontent.com/48173231/103893801-e568a100-5113-11eb-8044-1050f74a9511.png)

If I don't check for null then it thought error as using fn(...params) in merge function